src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1229,23 +1233,126 @@ private int SkipBlockComment(int i)
+                case 'r': return "\r";
+                case 't': return "\t";
+                case 'u': return ScanUnicodeEscapeSequence();
+                case 'v': return "\v";
+                default: return c.ToString();
+            }
+        }
+
+        private string ScanUnicodeEscapeSequence()
+        {
+            int escSeqStartIndex = _currentIndex - 2;
+            int maxNumberOfHexDigits = 6;
+            bool isTerminated = false;
+
+            char c = GetChar();
+            if (c != '{')
+            {
+                UngetChar();
+
+                IScriptExtent errorExtent = NewScriptExtent(escSeqStartIndex, _currentIndex);
+                ReportError(errorExtent, () => ParserStrings.InvalidUnicodeEscapeSequence);
+                return s_unknownUnicodeChar;
+            }
+
+            // Scan hex chars after the Unicode escape sequence start.
+            var sb = GetStringBuilder();
+            int i;
+            for (i = 0; i < maxNumberOfHexDigits; i++)
+            {
+                c = GetChar();
+
+                // Sequence has been terminated.
+                if (c == '}')
+                {
+                    if (i == 0)
+                    {
+                        // Sequence must have at least one hex char.
+                        IScriptExtent errorExtent = NewScriptExtent(escSeqStartIndex, _currentIndex);
+                        ReportError(errorExtent, () => ParserStrings.InvalidUnicodeEscapeSequence);
+                        return s_unknownUnicodeChar;
+                    }
+
+                    isTerminated = true;
+                    break;
+                }
+
+                if (!c.IsHexDigit())
+                {
+                    UngetChar();
+
+                    ReportError(_currentIndex, () => ParserStrings.MissingUnicodeEscapeSequenceTerminator);
+                    return s_unknownUnicodeChar;
+                }
+
+                sb.Append(c);
+            }
+
+            if (!isTerminated)
+            {
+                c = GetChar();
+                if (c != '}')
+                {
+                    UngetChar();
+                    ReportError(_currentIndex,
+                        i == maxNumberOfHexDigits && c.IsHexDigit()
+                            ? (Expression<Func<string>>)(() => ParserStrings.TooManyDigitsInUnicodeEscapeSequence)
+                            : () => ParserStrings.MissingUnicodeEscapeSequenceTerminator);
+                    return s_unknownUnicodeChar;
+                }
+            }
+
+            string hexStr = GetStringAndRelease(sb);
+
+            int unicodeValue = int.Parse(hexStr, NumberStyles.AllowHexSpecifier, NumberFormatInfo.InvariantInfo);
+            if (unicodeValue <= Char.MaxValue)
+            {
+                return ((char)unicodeValue).ToString();
+            }
+            else if (unicodeValue <= 0x10FFFF)
+            {
+                return Char.ConvertFromUtf32(unicodeValue);
```

Comment:
@rkeithhill 

> Try it on a different host e.g. ISE or VSCode.

Ah, good point, thanks. Is this just a limitation of the specific fonts available in the console?

> given that Bash, C# and Python don't disallow it.

(This is optional reading, in case you like that sort of thing. I'm not lobbying against the decision that was made.)

I'll address C# in a separate response.  

After digging a little deeper,  **regarding Bash/Ksh/Zsh and Python, as well as Ruby and Perl**:

* **Surrogate support is either _fundamentally broken_ or _disallowed_.**

   * **Fundamentally broken**:
       * `bash` (as of v4.4.12), `ksh` (as of v93u+), `zsh` (as of v5.2), `ruby` (v2.3-), `python` (3.x, as of 3.6.1)
       * All these either _mistakenly UTF-8-encode the individual surrogate_ code points - resulting in _invalid UTF-8_ - or report errors on trying to UTF-8 encode (which is a step in the right direction, because surrogate code points by themselves have no UTF-8 representation).
      * The problem stems from mistakenly treating the individual surrogate code points _as if they were valid characters by themselves_.
  * **Disallowed**:
      * `ruby` (v2.4), `perl` (as of at least v5.18)
      * Both complain on _parsing the string literal_ if it contains surrogate code points, though in the case of `perl` you just get _warnings_ (and you still get invalid UTF-8 as output).

* The **only exception is the - obsolescent - **Python 2.x****:
   * if you explicitly use Unicode string literals explicitly (`u"..."`)
   *  if you explicitly turn on UTF-8 support via environment variable `IOPYTHONENCODING`

   * Python 2.x _counts surrogate code points as individual characters_ (as, to my amazement, does C#).  
Python 3.3+ switched to a [new internal string representation](https://www.python.org/dev/peps/pep-0393/), which does _not_ use surrogate pairs internally, which now causes Unicode _characters_ to be counted correctly - but in the process the recognition of a surrogate _pair_ as a _single character_ was apparently broken.